pkg/registry/apis/provisioning/jobs/exporter.go

```diff
@@ -67,107 +71,106 @@ func (r *exporter) Export(ctx context.Context,
 	progress func(provisioning.JobStatus) error,
 ) (*provisioning.JobStatus, error) {
 	logger := logging.FromContext(ctx)
-	err := progress(provisioning.JobStatus{
+	status := provisioning.JobStatus{
 		State:   provisioning.JobStateWorking,
-		Message: "getting folder tree...",
-	})
-	if err != nil {
-		return nil, err
-	}
-	var ref string
-	if r.repository.Config().Spec.Type == provisioning.GitHubRepositoryType {
-		ref = r.repository.Config().Spec.GitHub.Branch
+		Message: "reading folder tree...",
 	}
-	ns := r.repository.Config().GetNamespace()
-	logger = logger.With("ref", ref, "namespace", ns)
-
-	err = progress(provisioning.JobStatus{
-		State:   provisioning.JobStateWorking,
-		Message: "exporting folders...",
-	})
+	err := progress(status)
 	if err != nil {
 		return nil, err
 	}
+	ref := options.Branch // only valid for git (defaults to the configured repo)
+	if options.Prefix != "" {
+		options.Prefix = safepath.Clean(options.Prefix)
+	}
 
 	// TODO: handle pagination
-	rawFolders, err := r.folders.List(ctx, metav1.ListOptions{})
+	rawFolders, err := r.folders.List(ctx, metav1.ListOptions{Limit: 10000})
 	if err != nil {
 		return nil, fmt.Errorf("failed to list folders: %w", err)
 	}
+	if rawFolders.GetContinue() != "" {
+		return nil, fmt.Errorf("unable to list all folders in one request: %s", rawFolders.GetContinue())
+	}
 
-	unprovisionedFolders := make([]unstructured.Unstructured, 0, len(rawFolders.Items))
-	for _, f := range rawFolders.Items {
-		repoName := f.GetAnnotations()[apiutils.AnnoKeyRepoName]
-		if repoName == repo.Config().GetName() {
-			logger.Info("skip as folder is already in repository", "folder", f.GetName())
```

Comment:
@MissingRoberto you asked what I meant by "hurt" the other day... this is a perfect example!

When exporting, we are writing files from grafana to a repository -- the state of grafana metadata is not relevant (we wipe it all out a few lines down!)

Reply:
similar behaviour with identifier = true

Reply:
notice the missing titles as well.

Reply:
Configuration used
```yaml
apiVersion: provisioning.grafana.app/v0alpha1
kind: Repository
metadata:
  name: github-ui-sync-demo
spec:
  title: Local
  description: load resources from local
  type: local
  sync:
    target: folder
    enabled: true
  local:
    path: devenv/dev-dashboards/panel-heatmap

```